src/components/PostDetails/PostDetails.tsx

```diff
@@ -1,45 +1,90 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { usePostsContext } from '../../customHooks/usePostsContext';
 import { NewCommentForm } from '../NewCommentForm';
+import { getPostDetails } from '../../api/posts';
 import './PostDetails.scss';
+import { Post } from '../../types/Post';
+import { deleteComment, getPostComments } from '../../api/comments';
 
-export const PostDetails: React.FC = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
+export const PostDetails: React.FC = React.memo(() => {
+  const { selectedPostId, comments, setComments } = usePostsContext();
+  const [postDetails, setPostDetails] = useState<Post>();
+  const [isCommentsVisible, setIsCommentsVisible] = useState(true);
 
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
+  const handleClick = () => {
+    setIsCommentsVisible((state) => !state);
+  };
 
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
+  const loadPostDetails = () => {
+    getPostDetails(selectedPostId)
+      .then(setPostDetails);
+  };
 
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
+  const loadPostComments = () => {
+    getPostComments(selectedPostId)
+      .then(setComments);
+  };
+
+  const toDeleteComment = (commentId = 0) => {
+    deleteComment(commentId);
+    const filteredComments = comments.filter(
+      comment => comment.id !== commentId,
+    );
+
+    setComments(filteredComments);
+  };
 
-        <li className="PostDetails__list-item">
+  useEffect(() => {
+    Promise.all([loadPostDetails(), loadPostComments()]);
```

Comment:
Для чого тут promise.all?
```suggestion
    loadPostDetails()
    loadPostComments();
```

Reply:
а почему нет ?